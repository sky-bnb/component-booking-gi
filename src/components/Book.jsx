import React from 'react';
import styled from 'styled-components';

export const StyledBookForm = styled.form`
`;

const StyledBookButton = styled.button`
  display: inline-block;
  position: relative;
  text-align: center;
  transition-property: background, border-color, color;
  transition-duration: 0.2s;
  transition-timing-function: ease-out;
  cursor: pointer;
  width: 100%;
  min-width: 71.1935px;
  margin: 0px;
  text-decoration: none;
  border-radius: 4px;
  border-width: 2px;
  border-style: solid;
  background: rgb(255, 90, 95);
  border-color: transparent;
  align-items: flex-start;
  box-sizing: border-box;
  border-image: initial;
  text-rendering: auto;
  word-spacing: normal;
  text-indent: 0px;
  text-shadow: none;
  font: 400 11px system-ui;
  margin: 0;
  padding: 10px 22px 10px 22px;

  :active {
    background: rgb(223, 60, 71);
  }

  :focus {
    outline: -webkit-focus-ring-color auto 5px;
  };
`;

const StyledButtonText = styled.div`
  overflow-wrap: break-word;
  font-family: 'Quicksand', sans-serif;
  font-size: 16px;
  font-weight: 600;
  line-height: 1.375em;
  color: rgb(255, 255, 255);
  margin: 0px;
`;

const StyledChargeText = styled.div`
  margin-top: 8px;
  font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, Helvetica Neue, sans-serif;
  font-size: 12px;
  font-weight: 600;
  line-height: 1.33333em;
  color: #484848;
  text-align: center;
  -webkit-font-smoothing: antialiased;
`;

const Book = () => (
  <StyledBookForm>
    <StyledBookButton>
      <StyledButtonText>Book</StyledButtonText>
    </StyledBookButton>
    <StyledChargeText>You won't be charged yet</StyledChargeText>
  </StyledBookForm>
);

export default Book;
