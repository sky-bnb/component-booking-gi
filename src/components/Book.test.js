import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Book, { StyledBookForm } from './Book.jsx';

describe('Book Component', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Book />));

  it('renders the Booking element', () => {
    expect(wrapper.find(StyledBookForm).length).toBe(1);
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
