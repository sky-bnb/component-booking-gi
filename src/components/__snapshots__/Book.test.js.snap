// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Component matches the snapshot 1`] = `
<Book__StyledBookForm>
  <Book__StyledBookButton>
    <Book__StyledButtonText>
      Book
    </Book__StyledButtonText>
  </Book__StyledBookButton>
  <Book__StyledChargeText>
    You won't be charged yet
  </Book__StyledChargeText>
</Book__StyledBookForm>
`;
